job-service-db/src/main/resources/procedures/processFailedDependentJobs.sql

```diff
@@ -36,13 +36,14 @@ BEGIN
         WITH RECURSIVE all_job_dependencies AS (
             SELECT partition_id, dependent_job_id, job_id
             FROM job_dependency
-            UNION ALL
+            WHERE partition_id = in_partition_id AND dependent_job_id = in_job_id
+            UNION
             SELECT adj.partition_id, adj.dependent_job_id, jd.job_id
             FROM all_job_dependencies adj
             INNER JOIN job_dependency jd ON adj.partition_id = jd.partition_id AND adj.job_id = jd.dependent_job_id
+            WHERE adj.partition_id = in_partition_id
         )
-        SELECT job_id FROM all_job_dependencies
-        WHERE dependent_job_id = in_job_id AND partition_id = in_partition_id;
+        SELECT DISTINCT job_id FROM all_job_dependencies
```

Comment:
`SELECT DISTINCT` over `SELECT`as even when using `UNION` there are still duplicate `job_ids` in the result set. So I've removed them here to make the rest of the function quicker.